### PR TITLE
Feature/43 | Added AutoScaling Same Az Elb plugin, spec file

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/autoscaling/sameAzElb.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            autoScalingGroups.data.forEach(function(asg) {
+                var asgAvailabilityZones = asg.AvailabilityZones;
+                var distinctAzs = [];
+                var resource = asg.AutoScalingGroupARN;
+
+                if(asg.HealthCheckType == 'ELB') {
+                    if (asg.LoadBalancerNames && asg.LoadBalancerNames.length) {
+
+                        asg.LoadBalancerNames.forEach(function(elbName) {
+                            if(elbName in loadBalancers) {
```
Comment (giorod3) [inline plugins/aws/autoscaling/sameAzElb.js:76]: Can you replace this check with `if(loadBalancers[elbName]) {` this technically does the same thing but is more aligned with the codebase

File: plugins/aws/autoscaling/sameAzElb.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                                var loadBalancer = loadBalancers[elbName];
+                                var elbAvailabilityZones = loadBalancer.AvailabilityZones;
+
+                                elbAvailabilityZones.forEach(function(elbAz) {
```
Comment (giorod3) [inline plugins/aws/autoscaling/sameAzElb.js:80]: before looping though elbAvailabilityZones check if it exists and has a length 
Suggestion: `if (elbAvailabilityZones && elbAvailabilityZones.length) {`

File: plugins/aws/autoscaling/sameAzElb.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                                    if(!asgAvailabilityZones.includes(elbAz)) {
```
Comment (giorod3) [inline plugins/aws/autoscaling/sameAzElb.js:81]: same thing here, make sure asgAvailabilityZones exists before doing .includes

**Review by giorod3 — CHANGES_REQUESTED** [review]

File: plugins/aws/autoscaling/sameAzElb.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Same Availability Zone in ASG and ELB',
```
Comment (matthewdfuller) [inline plugins/aws/autoscaling/sameAzElb.js:5]:
```suggestion
    title: 'AutoScaling ELB Same Availability Zone',
```

Comment (AkhtarAmir) [reply plugins/aws/autoscaling/sameAzElb.js]: Modified

**Review by giorod3 — APPROVED** [review]

**Pull request merged**